modal/_utils/git_utils.py

```diff
@@ -0,0 +1,75 @@
+# Copyright Modal Labs 2025
+import os
+import subprocess
+from typing import Dict, List, Optional, Union
+
+
+def run_command(args: List[str]) -> Optional[str]:
+    """Run a command and return its output.
+
+    Args:
+        args: Command and arguments as a list
+
+    Returns:
+        Command output as string or None if command failed
+    """
+    try:
+        result = subprocess.run(
+            args,
+            check=True,
+            capture_output=True,
+            text=True,
+            shell=False,
+        )
+        return result.stdout.strip()
+    except (subprocess.CalledProcessError, FileNotFoundError):
+        return None
+
+
+def get_git_commit_info() -> Dict[str, Union[str, bool, int]]:
+    """Collect git information about the current repository."""
+
+    # Check if we're in a git repository
+    if not os.path.exists(".git") and run_command(["git", "rev-parse", "--is-inside-work-tree"]) != "true":
+        return {}
+
+    git_info = {
+        "vcs": "git",
```

Comment:
I guess it's good to be proactive about supporting other VCSs although I worry that some of the data we're collecting is fairly git-specific. Problem we can solve later, I guess.